nifi-extension-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/DatabaseParameterProvider.java

```diff
@@ -38,37 +26,34 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.ServiceLoader;
 import java.util.stream.Collectors;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.ConfigVerificationResult;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.PropertyDescriptor.Builder;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.db.DatabaseAdapter;
+import org.apache.nifi.dbcp.DBCPService;
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.util.StringUtils;
 
 @Tags({"database", "dbcp", "sql"})
 @CapabilityDescription("Fetches parameters from database tables")
 
 public class DatabaseParameterProvider extends AbstractParameterProvider implements VerifiableParameterProvider {
 
-    protected final static Map<String, DatabaseAdapter> dbAdapters = new HashMap<>();
-
-    public static final PropertyDescriptor DB_TYPE;
-
-    static {
-        // Load the DatabaseAdapters
-        ArrayList<AllowableValue> dbAdapterValues = new ArrayList<>();
-        ServiceLoader<DatabaseAdapter> dbAdapterLoader = ServiceLoader.load(DatabaseAdapter.class);
-        dbAdapterLoader.forEach(it -> {
-            dbAdapters.put(it.getName(), it);
-            dbAdapterValues.add(new AllowableValue(it.getName(), it.getName(), it.getDescription()));
-        });
-
-        DB_TYPE = new PropertyDescriptor.Builder()
-                .name("db-type")
-                .displayName("Database Type")
-                .description("The type/flavor of database, used for generating database-specific code. In many cases the Generic type "
-                        + "should suffice, but some databases (such as Oracle) require custom SQL clauses. ")
-                .allowableValues(dbAdapterValues.toArray(new AllowableValue[dbAdapterValues.size()]))
-                .defaultValue("Generic")
-                .required(true)
-                .build();
-    }
+    // TODO: There was no migration facility for Parameter Providers by the time of introducing this property,
```

Comment:
better to have todo not in the code and instead just file the JIRA for this topic if necessary

Reply:
Alright, I've filed a jira for this: https://issues.apache.org/jira/browse/NIFI-13327

Reply:
What would be the good strategy, given that there is no way to apply migration on Parameter Providers?

1. Have the fixed list of Db adapters in the {{DatabaseParameterProvider}}?
2. Replace the fixed list with the new controller service concept, and require a manual migration step from the user after upgrade?
3. Block this commit until the issue is resolved?

Reply:
@joewitt , @exceptionfactory 

Do you have any thoughts regarding to this?

Reply:
@joewitt , @exceptionfactory 

I have a draft PR that attempts to add the missing, property migration feature to the Parameter Provider component. Currently it is only half-complete, becuase I can't really find where is the proper place to call the `migrateProperties()` method. The place where I've now inserted it, seems to not get called. I assume it is because Parameter Providers do not belong to Process Groups.

Could you please have a look?

[Pull Request #9080](https://github.com/apache/nifi/pull/9080)